### ISSUE TITLE
HIP package only works on AMD platform, not on NVIDIA

### ISSUE THREAD

**Issue by jakub-homola** [LLNL/spack#26898]
The HIP package seems to only support AMD platform, on NVIDIA platform the HIP tools don't work correctly (because they are set up for AMD).

Before installing the hip package through spack, I made sure the `HIP_PLATFORM` environment variable was set to `nvidia`, but it did not have any effect on the installation.

The `module load hip` command overwrites the `HIP_PLATFORM` environment variable to `amd`, even though I amd on nvidia. If I set it back to `nvidia` and try to compile something using `hipcc`, another error appears: `nvcc fatal   : Unknown option '--rocm-path=/home/jakub/apps/spack/opt/spack/linux-ubuntu18.04-zen2/gcc-10.1.0/hip-4.3.1-shn6kb45egbifwo36efh3nyfqhcczr4y'`. The core of this is the `HIPCC_COMPILE_FLAGS_APPEND` environment variable, which was set by the module load to `--rocm-path=/home/jakub/apps/spack/opt/spack/linux-ubuntu18.04-zen2/gcc-10.1.0/hip-4.3.1-shn6kb45egbifwo36efh3nyfqhcczr4y`. Manually clearing the variable solved the issue and `hipcc` works, but this is not a very nice way to make it work.

To sum up, the two main issues are:
- unnecessary dependencies (AMD-specific stuff)
- incorrect environment setup when loading the module

Please correct this. HIP is supposed to work on **both** AMD and NVIDIA platform. I can suggest making two distinct packages like hip_nvidia and hip_amd. Steering the installation process based on the `HIP_PLATFORM` does not seem like a good idea.

I did not test any hip* libraries like hipBLAS, hipSPARSE etc., but it can be assumed that the issue will be similar with them.

**Comment by npf** [comment]
Furthermore, hip does not build on a machine with no AMD GPU:
```
==> Installing hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk
==> No binary for hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/ce/ce755ee6e407904eba3f6b3c9efcdd48eb4f58a26b06e1892166d05f19a75973.tar.gz
==> Fetching https://mirror.spack.io/_source-cache/archive/70/707f2217f0e7aeb62d7b76830a271056d665542bf5f7a54e40adf4d5f299ca93.tar.gz
==> Fetching https://mirror.spack.io/_source-cache/archive/44/44a7fac721abcd93470e1a7e466bdea0c668c253dee93e4f1ea9a72dbce4ba31.tar.gz
==> Fetching https://mirror.spack.io/_source-cache/archive/dd/ddee63cdc6515c90bab89572b13e1627b145916cb8ede075ef8446cbb83f0a48.tar.gz
==> Moving resource stage
	source: /tmp/root/spack-stage/resource-hipamd-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src/
	destination: /tmp/root/spack-stage/spack-stage-hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src/hipamd
==> Moving resource stage
	source: /tmp/root/spack-stage/resource-opencl-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src/
	destination: /tmp/root/spack-stage/spack-stage-hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src/opencl
==> Moving resource stage
	source: /tmp/root/spack-stage/resource-rocclr-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src/
	destination: /tmp/root/spack-stage/spack-stage-hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src/rocclr
==> Applied patch /grid5000/spack/develop/var/spack/repos/builtin/packages/hip/0011-Improve-compilation-without-git-repo-and-remove-compiler-rt-linkage-for-host.5.0.2.patch
==> Ran patch() for hip
==> hip: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/lib64;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/comgr-5.1.3-dqyk4xtpkzoj6yhg7d2stvgadtiy7tk6/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/llvm-amdgpu-5.1.3-xitxos7wptlhjruxsqbluoiupk3jsnnz/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/ncurses-6.2-n4fq345wdbt2mhj6tw23f7ykpiqyj77j/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/z3-4.8.16-ac4j23hdbxm25klg5k5kx46tofsqbm3z/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/zlib-1.2.12-ta6zb2hlf5h5fsyuqi5tkqjasommierg/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/hsa-rocr-dev-5.1.3-7w26qenvwu3fwlb2mg2bnpyb2gz7d53l/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/elfutils-0.186-gfr4xvo3a5ct7tqf37gehjpspy25h3h2/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/gettext-0.21-kriycqwqjsnn33iu422jrvesfdmuru23/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/bzip2-1.0.8-h4q5eq2l2v732i6f3cu6uo7bweqxyqi2/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libiconv-1.16-zrrzcnmurb2zstx4kf4xfrxr7udbh4df/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxml2-2.9.13-jkgdsouxmg4gxj6gerfquoeznatfxpm5/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/xz-5.2.5-wtz7eaotekjt6wywbtfg6hwhqoyqt5vl/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/hsakmt-roct-5.1.3-3cythjnwbvxn2pac2j4e5hdophiw7sqo/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libdrm-2.4.110-ikgz2d36a6cur6zqccyngcmdpbsjgvcw/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libpciaccess-0.16-up5lbgjf43vduijwkapweelyavatizkf/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libpthread-stubs-0.4-renz33ee5sjdk2vnxznbs7a4osbb773p/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/numactl-2.0.14-m462qep2wfagujquna6tagstr7yrryoi/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/mesa-22.0.2-kzzzcikmpa6rofuqlw2u3lnoc6ql5dd3/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/expat-2.4.8-oluynsvgpdbhcgucucyf2eiqet6pd55e/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libbsd-0.11.5-ytpt6soe5lq7bstlpbamr64mc4inmomt/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libmd-1.0.4-qrv2ckdgpglpcebmgjfqkclyjejg6pqj/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/glproto-1.4.17-3gqzazmqafyeu5j3qw7hk2fplfe5wpfg/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libunwind-1.6.2-io5wcei5cmezgs5wye2mtqacj6sf6nqj/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libx11-1.7.0-6p44fq5wzaogpdpcvzhopsgcico4ljuu/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/inputproto-2.3.2-d5wvtuvlrbgedg2mcvvfoukugafvojee/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/kbproto-1.0.7-ihpmwlqg4ntm7ygxuuxadclgsb6mjhjo/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxcb-1.14-32pte6lz4w2lutzsemkpatemxap3efhq/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxau-1.0.8-f6q5sjgffb4wjyzwqgqdki3i2eoqg3vc/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/xproto-7.0.31-chusyshhiqstf7cuhfs6kdg6wxoqjd5n/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxdmcp-1.1.2-yxm7f7xvhuevw3wdngo4i6m4hckimvlk/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/xcb-proto-1.14.1-zs3duayr5wuh4qs5h67vlo22dgsmdici/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/xextproto-7.3.0-tk5u6jsbjdzvwn4usxeotwalarjjqpyu/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxext-1.3.3-le2e56bcsww7nw4f4pmehbx7pokivi4n/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxt-1.1.5-7sq6qxtjoiylu2u2l6tgotcxuqscoago/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libice-1.0.9-pbw2fivornalqhyzil46focky4bu4hkn/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libsm-1.2.3-auwomqu7iia53xxoiyi56kzarpmbapvs/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/util-linux-uuid-2.37.4-ewicryuu35rz7lro7bkix44ul7rgbmtx/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxrandr-1.5.0-nbz437mvk6tzkoslgmje47zlpa7s45xq/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxrender-0.9.10-wsgqiywthgwgjmf3peutbs7avxwsbvze/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/renderproto-0.11.1-6femmtr23flc5jn2k5ezsi3l3mgbtftf/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/randrproto-1.5.0-ubghjksbklsjuyw6nw6kltlxrnhir7ek/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/perl-file-which-1.22-sfi44fnps6nosvc6rhglhyjeczahwxvs/lib;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/perl-uri-encode-1.1.1-iplvipyl74ct2mbkgmiynnlpalh2eahg/lib' '-DCMAKE_PREFIX_PATH:STRING=/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/roctracer-dev-api-5.1.3-uu7xe4ea4bdjnz6vvxe53xpdnrecvxv5;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/rocminfo-5.1.3-md2boz43dg3nidggozgmkq4s5thsnnru;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/perl-uri-encode-1.1.1-iplvipyl74ct2mbkgmiynnlpalh2eahg;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/perl-file-which-1.22-sfi44fnps6nosvc6rhglhyjeczahwxvs;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/mesa-22.0.2-kzzzcikmpa6rofuqlw2u3lnoc6ql5dd3;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/xrandr-1.5.0-3srklgw7pz4sqkxtvyueumyynqhqnhbi;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxrandr-1.5.0-nbz437mvk6tzkoslgmje47zlpa7s45xq;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/randrproto-1.5.0-ubghjksbklsjuyw6nw6kltlxrnhir7ek;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxrender-0.9.10-wsgqiywthgwgjmf3peutbs7avxwsbvze;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/renderproto-0.11.1-6femmtr23flc5jn2k5ezsi3l3mgbtftf;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxt-1.1.5-7sq6qxtjoiylu2u2l6tgotcxuqscoago;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libsm-1.2.3-auwomqu7iia53xxoiyi56kzarpmbapvs;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libice-1.0.9-pbw2fivornalqhyzil46focky4bu4hkn;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxext-1.3.3-le2e56bcsww7nw4f4pmehbx7pokivi4n;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libx11-1.7.0-6p44fq5wzaogpdpcvzhopsgcico4ljuu;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/xtrans-1.3.5-3raevrwqtfvnka6ec7ls6wmwvnkagr4q;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/xextproto-7.3.0-tk5u6jsbjdzvwn4usxeotwalarjjqpyu;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxcb-1.14-32pte6lz4w2lutzsemkpatemxap3efhq;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/xcb-proto-1.14.1-zs3duayr5wuh4qs5h67vlo22dgsmdici;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxdmcp-1.1.2-yxm7f7xvhuevw3wdngo4i6m4hckimvlk;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxau-1.0.8-f6q5sjgffb4wjyzwqgqdki3i2eoqg3vc;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/xproto-7.0.31-chusyshhiqstf7cuhfs6kdg6wxoqjd5n;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/kbproto-1.0.7-ihpmwlqg4ntm7ygxuuxadclgsb6mjhjo;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/inputproto-2.3.2-d5wvtuvlrbgedg2mcvvfoukugafvojee;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libunwind-1.6.2-io5wcei5cmezgs5wye2mtqacj6sf6nqj;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/glproto-1.4.17-3gqzazmqafyeu5j3qw7hk2fplfe5wpfg;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/hsa-rocr-dev-5.1.3-7w26qenvwu3fwlb2mg2bnpyb2gz7d53l;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/hsakmt-roct-5.1.3-3cythjnwbvxn2pac2j4e5hdophiw7sqo;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/numactl-2.0.14-m462qep2wfagujquna6tagstr7yrryoi;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libdrm-2.4.110-ikgz2d36a6cur6zqccyngcmdpbsjgvcw;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libpthread-stubs-0.4-renz33ee5sjdk2vnxznbs7a4osbb773p;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libpciaccess-0.16-up5lbgjf43vduijwkapweelyavatizkf;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/elfutils-0.186-gfr4xvo3a5ct7tqf37gehjpspy25h3h2;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/comgr-5.1.3-dqyk4xtpkzoj6yhg7d2stvgadtiy7tk6;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/llvm-amdgpu-5.1.3-xitxos7wptlhjruxsqbluoiupk3jsnnz;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/z3-4.8.16-ac4j23hdbxm25klg5k5kx46tofsqbm3z;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/util-linux-uuid-2.37.4-ewicryuu35rz7lro7bkix44ul7rgbmtx;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/gettext-0.21-kriycqwqjsnn33iu422jrvesfdmuru23;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/tar-1.34-jh4mk5ewc3vwisopwefjqa7hftbdeok6;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libxml2-2.9.13-jkgdsouxmg4gxj6gerfquoeznatfxpm5;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/xz-5.2.5-wtz7eaotekjt6wywbtfg6hwhqoyqt5vl;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/expat-2.4.8-oluynsvgpdbhcgucucyf2eiqet6pd55e;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libbsd-0.11.5-ytpt6soe5lq7bstlpbamr64mc4inmomt;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libmd-1.0.4-qrv2ckdgpglpcebmgjfqkclyjejg6pqj;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/cmake-3.23.2-yivam3sbh6lrqwc36nf65kolvwi65zzx;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/perl-5.34.1-dh2m2r6hqzkiunozukue5wygc7kb5nux;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/zlib-1.2.12-ta6zb2hlf5h5fsyuqi5tkqjasommierg;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/bzip2-1.0.8-h4q5eq2l2v732i6f3cu6uo7bweqxyqi2;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/libiconv-1.16-zrrzcnmurb2zstx4kf4xfrxr7udbh4df;/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/ncurses-6.2-n4fq345wdbt2mhj6tw23f7ykpiqyj77j' '-DPROF_API_HEADER_PATH:STRING=/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/roctracer-dev-api-5.1.3-uu7xe4ea4bdjnz6vvxe53xpdnrecvxv5/roctracer/inc/ext' '-DHIP_COMPILER:STRING=clang' '-DHSA_PATH:STRING=/grid5000/spack/develop/opt/spack/linux-debian11-x86_64_v2/gcc-10.3.0/hsa-rocr-dev-5.1.3-7w26qenvwu3fwlb2mg2bnpyb2gz7d53l' '-DHIP_RUNTIME:STRING=rocclr' '-DHIP_PLATFORM:STRING=amd' '-DHIP_COMMON_DIR:STRING=/tmp/root/spack-stage/spack-stage-hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src' '-DHIP_CATCH_TEST:STRING=OFF' '-DROCCLR_PATH:STRING=/tmp/root/spack-stage/spack-stage-hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src/rocclr' '-DAMD_OPENCL_PATH:STRING=/tmp/root/spack-stage/spack-stage-hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src/opencl' '/tmp/root/spack-stage/spack-stage-hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src/hipamd'

2 errors found in build log:
     54    -- Found HIP: /tmp/root/spack-stage/spack-stage-hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-build-ma6okxn (found version "5.1.0")
     55    -- host interface - found
     56    -- HIP runtime lib type - shared
     57    -- CMAKE_TESTING_TOOL:
     58    -- CMAKE HIP ARCHITECTURES: OFF
     59    -- ROCm Agent Enumurator Result: 0
  >> 60    CMake Error at /tmp/root/spack-stage/spack-stage-hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src/tests/hit/HIT.cmake:51 (string):
     61      string sub-command LENGTH requires two arguments.
     62    Call Stack (most recent call first):
     63      CMakeLists.txt:446 (include)
     64    
     65    
  >> 66    CMake Error at /tmp/root/spack-stage/spack-stage-hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-src/tests/hit/HIT.cmake:54 (if):
     67      if given arguments:
     68    
     69        "GREATER_EQUAL" "1"
     70    
     71      Unknown arguments specified
     72    Call Stack (most recent call first):

See build log for details:
  /tmp/root/spack-stage/spack-stage-hip-5.1.3-ma6okxn4xtlyf7rhbw6cfqwlh6rufihk/spack-build-out.txt


Failed.
```
The problem comes in `tests/hit/HIT.cmake` from that `HIP_PLATFORM` is set to `amd`, while `HIP_GPU_ARCH` cannot be guessed, thus ends up null.

* **Spack:** 0.19.0.dev0 (f06aa9e785c78cb6f532337ff4b0660a9d00dd5a)
* **Python:** 3.9.2
* **Platform:** linux-debian11-skylake_avx512
* **Concretizer:** clingo

**Comment by srekolam** [comment]
#33872 is now merged. Let us know if this can be closed. Thanks.

**Comment by jakub-homola** [comment]
I'm not working with HIP for some time. If the PR fixes the issue, you can close it.

**Comment by srekolam** [comment]
Thanks.  Closing this issue as the PR is already merged.
